### PR TITLE
Story Aggregation 

### PR DESCRIPTION
**Pull request by codelicous**
Saving content in array - moving story state to game context

### CODE REVIEW

File: src/components/game-board/GameBoard.tsx [hunk 1]
```diff
@@ -55,52 +47,53 @@ function GameBoard({ className }: ChildProps): React.JSX.Element{
...
+            activePlayer: prevGame.players[nextPlayerIndex],
+            nextPlayer: prevGame.players[(nextPlayerIndex + 1) % prevGame.players.length],
 
-            };
-        });
-    }, []);
+        };
+    },[]);
+    const updatePlayerTurn = useCallback(() => {
+        setShowGameDialog(false);
```
Comment (ofergroman) [inline src/components/game-board/GameBoard.tsx:65]: Is this really relevant to "updatePlayerTurn"? Isn't this more relevant to a "startGame" function?

File: src/components/game-board/story-board/StoryBoard.tsx [hunk 2]
```diff
@@ -44,7 +52,7 @@ export default function StoryBoard({className, content, updatePlayerTurn, active
     return <div className={className}>
         <div className='flex flex-col h-3/4 w-full items-center'>
             <div className='text-container w-full border-2 h-3/4 flex flex-1 text-xl p-5'>
-                <div>{submitted}</div>
+                <div className='ho ken'>{content}</div>
```
Comment (ofergroman) [inline src/components/game-board/story-board/StoryBoard.tsx:55]: like

File: src/components/game-board/GameBoard.tsx (hunk 1 above)
Comment (codelicous) [reply src/components/game-board/GameBoard.tsx:65]: you are right , its redundant here

File: src/components/game-board/GameBoard.tsx [hunk 3]
```diff
@@ -43,32 +37,34 @@ function GameBoard({ className }: ChildProps): React.JSX.Element{
...
-            };
-        });
-    }, []);
+    const updatePlayerTurn = useCallback(() => {
+        setGame(updatePlayerInsideGameObject);
+    }, [updatePlayerInsideGameObject]);
 
     // useEffect to initialize the game
     useEffect(() => {
+        if(showGameDialog) {
```
Comment (e1ad) [inline src/components/game-board/GameBoard.tsx:59]: This is a side effect of action done by the GameDialog, I think we can move it to there since we have access to setGame from anywhere. We wont need the useEffect but a callBack function the will be called when the the dialog is getting closed.

File: src/contexts/game.context.tsx [hunk 4]
```diff
@@ -16,6 +34,15 @@ type GameProps = { children?: ReactNode };
 
 export const GameProvider: React.FC<GameProps> = ({ children }) => {
     const [isTriggered, setIsTriggered] = useState(false);
+    const [story,setStory] = useState<Story>({entries: [], opener:''});
+    const [content, setContent] = useState('');
+    const addEntry = useCallback((storyEntry: StoryEntry) => {
+        setStory({...story, entries: [...story.entries, storyEntry]});
```
Comment (e1ad) [inline src/contexts/game.context.tsx:40]: As we talked about, lets please have single object name gameData to store all game's dynamic data.
